### ISSUE TITLE
update issue : file not found xlwings32-0.15.1.dll

### ISSUE THREAD

**Issue by Hmissa** [xlwings/xlwings#1070]
#### OS (e.g. Windows 10 or macOS Sierra)
windows 7
#### Versions of xlwings, Excel and Python (e.g. 0.11.8, Office 365, Python 3.7)
from xlwings 0.15.1 to xlwings 0.15.5
#### Describe your issue (incl. Traceback!)
python 3.7

I have a excel file that use xlwings 0.15.1, I upgrade the package and the addin to 0.15.5

when I run a function imported from the python script I have an error message File note found: xlwings32-0.15.1.dll

How Can I make the xlwings point to the new dll file xlwings32-0.15.5.dll

Best Regards,
Saif.

**Comment by fzumstein** [comment]
Something must have gone wrong then with the addin upgrade, does it show Version: 0.15.5?

**Comment by Hmissa** [comment]
I found a solution:
I opened the xlwings module and I replaced the  xlwings64-0.15.1.dll and xlwings32-0.15.1.dll by xlwings64-0.15.5.dll and xlwings32-0.15.5.dll

Regards,
Saif.

**Comment by akshay1205** [comment]
Where do I find xlwings64-0.15.5.dll and xlwings32-0.15.5.dll? I am having the same error and need to download these for debugging the error.

**Comment by fzumstein** [comment]
what version do you see in the ribbon?

**Comment by akshay1205** [comment]
0.15.5

**Comment by akshay1205** [comment]
<img width="749" alt="VBA error" src="https://user-images.githubusercontent.com/49674777/56225056-4a25c300-6036-11e9-81e0-18ad65da0cce.PNG">

**Comment by Hmissa** [comment]
Hi,

Just you have to go to the xlwings vba module in Excel file:
![captur1](https://user-images.githubusercontent.com/7750324/56231861-5aaa5d80-6077-11e9-9864-5f33f703798e.jpg)


and update the dll file name

![captur](https://user-images.githubusercontent.com/7750324/56231880-672eb600-6077-11e9-856f-170e6ddca081.jpg)


Regards,
Saif.

**Comment by fzumstein** [comment]
Please don't do that manually as you'll miss out all the other changes. Instead, go to your installation folder of xlwings and import the`xlwings.bas` file from there. do `import xlwings` and `xlwings.__path__` to find your installation directory.

**Comment by akshay1205** [comment]
I am sorry but I don't have much experience in VBA coding. Do you mean to say I should use import xlwings and xlwings._path_ in VBA? and should I use both of them within the sub module? Or should I keep it global?
<img width="960" alt="VBA error1" src="https://user-images.githubusercontent.com/49674777/56243071-51ad9200-605f-11e9-9311-28982bd144b8.PNG">

**Comment by fzumstein** [comment]
you're using the addin, so it doesn't apply to you. Rather you will need to fix your python xlwings installation: Shut down Excel, then run `pip install xlwings==0.15.5`

**Comment by akshay1205** [comment]
<img width="651" alt="python xlwings" src="https://user-images.githubusercontent.com/49674777/56255270-e5945380-6089-11e9-95eb-6bb087f664a0.PNG">
I am getting this as I enter the above command in the prompt. Should I upload the python part of the xlwings code in here to understand the error?

**Comment by akshay1205** [comment]
I solved the above error, but I have a new error. I have uploaded the image below. Please help me out with this problem.
<img width="958" alt="2nderror" src="https://user-images.githubusercontent.com/49674777/56256641-3490b780-608f-11e9-8eb1-d6bd1b9c389c.PNG">

**Comment by akshay1205** [comment]
@fzumstein  Please help me to debug this error. I would be very grateful if you could help me do so.

**Comment by fzumstein** [comment]
What are you doing that raises this error?

**Comment by akshay1205** [comment]
I am working on a project where I am building a transportation model using linear programming on python. I am using Gurobi optimizer as my package and I am importing data from an excel sheet. I am trying to put my result back in the excel through xlwings package. I can show you a snip of my python code is required.
_Python Code:-_
def run():
    wb = xw.Book.caller()
    
    df13 = DataFrame({'flows':[flow1var,flow2var,'Total Transportation Cost'], 'volume':[flow1val,flow2val,totalcost]})
    wb.sheets['Result'].range("A1").value = df13

_Excel VBA code:-_
Sub my_transp()
RunPython ("import transp ; transp.run()")
End Sub

**Comment by fzumstein** [comment]
`run` is a reserved function name in VBA - rename your function and try again

**Comment by akshay1205** [comment]
I am still getting the same error as above. Has it something to do with putting a check on RunPython: Use UDF server under xlwings tab in excel workbook
<img width="586" alt="xlwings" src="https://user-images.githubusercontent.com/49674777/56322802-9dc80780-612f-11e9-9635-a25d9e904f40.PNG">
?

**Comment by akshay1205** [comment]
If I uncheck that, and run the VBA, I am getting module not found: Cannot find gurobi module.

**Comment by fzumstein** [comment]
can you import gurobi in a command prompt (not an anaconda prompt)? after starting python by typing `python`?

**Comment by akshay1205** [comment]
I will try that.

**Comment by gregchamblin** [comment]
This helped me too. Maybe there is an opportunity for better error readability?

**Comment by fzumstein** [comment]
Can you be a bit more specific @gregchamblin? This issue is a mix of things.

**Comment by gregchamblin** [comment]
@fzumstein I apologize, I know better than to not give details. I started out with multiple errors, Error 1000, file not found (xlwings64-0.19.1.dll), etc. Putting a check on "RunPython: Use UDF Server" box told me what module was missing and I installed it. Problem solved.

It has been 7 weeks since I last used this sheet. I moved to python 3.8 (I did not reinstall all packages from 3.7 on purpose) and upgraded to 0.19.1. 

I understand that I broke it, it just seems that I wouldn't have spent an hour fixing it if I had a clear/more precise error message up front.
